Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/Commands.cpp

```diff
@@ -319,6 +320,21 @@ namespace EMStudio
                 // Add motion meta data.
                 auto motionMetaData = AZStd::make_shared<EMotionFX::Pipeline::Rule::MotionMetaData>(motion->GetMotionExtractionFlags(), motion->GetEventTable());
                 EMotionFX::Pipeline::Rule::SaveToGroup<EMotionFX::Pipeline::Rule::MotionMetaDataRule, AZStd::shared_ptr<EMotionFX::Pipeline::Rule::MotionMetaData>>(*scene, group, motionMetaData);
+
+                // Save RootMotionExtractionRule
+                auto rootMotionData = motion->GetRootMotionExtractionData();
+                if (rootMotionData)
```

Comment:
nit: We could update this to

```c++
if (auto rootMotionData = motion->GetRootMotionExtractionData())
```

or

```c++
if (auto rootMotionData = motion->GetRootMotionExtractionData(); rootMotionData != nullptr)
```

Reply:
I will change this one. I'm too old school.